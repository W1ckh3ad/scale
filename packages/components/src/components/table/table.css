.table {
  display: block;
  overflow: auto;
  border-radius: var(--scl-radius-4) var(--scl-radius-4) 0 0;
  background: var(--scl-color-background-light);
  color: var(--scl-color-text-default);
}
.table table {
  width: 100%;
  white-space: nowrap;
  border-spacing: 0;
  border-collapse: collapse;
}
.table th {
  padding: 0 8px;
  text-align: left;
  line-height: 32px; /*  TODO: revise, this is weird */
  color: var(--scl-color-text-default);
  font-size: var(--scl-font-size-12);
}
.table td {
  padding: 0 8px;
}
.table tbody tr td {
  padding: 8px;
}
.table tbody td {
  border-bottom: 1px solid var(--scl-color-grey-20);
}

.table tfoot,
.table tbody {
  background: var(--scl-color-background-default);
}
.table tfoot tr td {
  padding: 8px;
  font-weight: 600;
  border-bottom: 1px solid var(--scl-color-text-default);
}
.table .scale-sort-indicator svg {
  display: none;
}
.table tbody tr:hover {
  background: #ededed;
}
.table th:focus {
  outline: none;
}
.table--size-default {
}
.table--size-default tbody tr td {
  padding: 16px 8px;
}
.table--sortable {
}
.table--sortable th {
  cursor: pointer;
  padding: 0 8px 0 0;
  border-radius: var(--scl-radius-4) var(--scl-radius-4) 0 0;
}
.table--sortable th:hover {
  padding: 0 8px 0 0;
  background: #ededed;
}
.table--sortable th:active {
  background: #dcdcdc;
}
.table--sortable th:focus {
  box-shadow: inset var(--scl-shadow-focus-default);
}
.table--sortable th .scale-sort-indicator svg {
  display: block;
}
.table--sortable th[aria-disabled] {
  padding: 0 8px;
  pointer-events: none;
}
.table--sortable th:not([aria-sort]) {
  padding: 0 8px;
}
.table--sortable th[aria-sort='none'] {
  padding: 0 8px;
}
.table--sortable
  th[aria-sort='ascending']
  .scale-sort-indicator
  polygon:first-of-type {
  fill: var(--scl-color-text-default);
}
.table--sortable th:hover .scale-sort-indicator polygon:first-of-type {
  fill: var(--scl-color-primary-hover);
}
.table--sortable
  th:hover[aria-sort='ascending']
  .scale-sort-indicator
  polygon:first-of-type {
  fill: var(--scl-color-primary-hover);
}
.table--sortable
  th:active[aria-sort='ascending']
  .scale-sort-indicator
  polygon:first-of-type {
  fill: var(--scl-color-primary-active);
}
.table--sortable th[aria-sort='ascending'] .scale-sort-indicator polygon {
  fill: transparent;
}
.table--sortable th:hover[aria-sort='ascending'] .scale-sort-indicator polygon {
  fill: transparent;
}
.table--sortable
  th:active[aria-sort='ascending']
  .scale-sort-indicator
  polygon {
  fill: transparent;
}
.table--sortable
  th[aria-sort='descending']
  .scale-sort-indicator
  polygon:first-of-type {
  fill: transparent;
}
.table--sortable
  th:hover[aria-sort='descending']
  .scale-sort-indicator
  polygon:first-of-type {
  fill: transparent;
}
.table--sortable
  th:active[aria-sort='descending']
  .scale-sort-indicator
  polygon:first-of-type {
  fill: transparent;
}
.table--sortable th[aria-sort='descending'] .scale-sort-indicator polygon {
  fill: var(--scl-color-text-default);
}
.table--sortable
  th:hover[aria-sort='descending']
  .scale-sort-indicator
  polygon {
  fill: var(--scl-color-primary-hover);
}
.table--sortable
  th:active[aria-sort='descending']
  .scale-sort-indicator
  polygon {
  fill: var(--scl-color-primary-active);
}
.table--sortable th .scale-sort-indicator {
  width: 16px;
  height: 16px;
  margin: 0 0 0 4px;
  display: inline-block;
  position: relative;
}
.table--sortable th .scale-sort-indicator > svg {
  top: 4px;
  left: 0;
  width: 16px;
  height: 16px;
  position: absolute;
}
.table--sortable th .scale-sort-indicator polygon {
  fill: transparent;
}
.table--sortable th[aria-sort='none']:hover {
  padding: 0 8px 0 0;
}
.table--sortable th[aria-sort='none'] .scale-sort-indicator {
  display: none;
}
.table--sortable th:not([aria-sort]):hover {
  padding: 0 8px 0 0;
}
.table--sortable th:not([aria-sort]) .scale-sort-indicator {
  display: none;
}
.table--sortable th:hover .scale-sort-indicator {
  display: inline-block;
}
