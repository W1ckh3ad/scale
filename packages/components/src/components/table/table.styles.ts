import { JssStyle } from 'jss';

export const styles: JssStyle = {
  table: {
    display: 'block',
    overflow: 'auto',
    borderRadius: '4px 4px 0 0',
    background: ({ colors }) => colors.grey[100],
    color: ({ color }) => color.text,

    '& table': {
      borderSpacing: 0,
      borderCollapse: 'collapse',
      width: '100%',
      whiteSpace: 'nowrap',
    },

    '& th': {
      color: ({ color }) => color.text,
      lineHeight: '32px', // TODO revise, this is weird
      fontSize: ({ type }) => type.size_2,
      textAlign: 'left',
      userSelect: 'none',
      padding: '0 8px',
      '&:focus': {
        outline: 'none',
      },
    },
    '& td': {
      padding: '0 8px',
    },

    '& tbody tr td': {
      padding: '8px',
    },
    '& tbody td': {
      borderBottom: ({ color, size }) =>
        `${size.divider}px solid ${color.divider}`,
    },

    '& tbody tr': {
      '&:hover': {
        background: '#EDEDED',
      },
    },

    '& tfoot, tbody': {
      background: '#FFFFFF',
    },

    '& tfoot tr td': {
      fontWeight: 600,
      padding: '8px',
      borderBottom: ({ color }) => `1px solid ${color.text}`,
    },

    '& .scale-sort-indicator svg': {
      display: 'none',
    },
  },

  'table--size-default': {
    '& tbody tr td': {
      padding: '16px 8px',
    },
  },

  'table--sortable': {
    '& th': {
      cursor: 'pointer',
      borderRadius: '4px 4px 0 0',
      padding: '0 8px 0 0',

      '&:hover': {
        '& .scale-sort-indicator': {
          display: 'inline-block',
        },
        padding: '0 8px 0 0',
        background: '#EDEDED',
      },

      '&:active': {
        background: '#DCDCDC',
      },

      '&:focus': {
        boxShadow: ({ color }) => `0 0 0 2px ${color.focus} inset`,
      },

      '& .scale-sort-indicator svg': {
        display: 'block',
      },
      '&[aria-disabled]': {
        pointerEvents: 'none',
        padding: '0 8px',
      },

      '&:not([aria-sort])': {
        padding: '0 8px',
        '&:hover': {
          padding: '0 8px 0 0',
        },
        '& .scale-sort-indicator': {
          display: 'none',
        },
      },
      '&[aria-sort="none"]': {
        padding: '0 8px',
        '&:hover': {
          padding: '0 8px 0 0',
        },
        '& .scale-sort-indicator': {
          display: 'none',
        },
      },

      '&[aria-sort="ascending"] .scale-sort-indicator polygon:first-of-type': {
        fill: ({ color }) => color.text,
      },
      '&:hover .scale-sort-indicator polygon:first-of-type': {
        fill: ({ color }) => color.primary_hover,
      },
      '&:hover[aria-sort="ascending"] .scale-sort-indicator polygon:first-of-type': {
        fill: ({ color }) => color.primary_hover,
      },
      '&:active[aria-sort="ascending"] .scale-sort-indicator polygon:first-of-type': {
        fill: ({ color }) => color.primary_active,
      },

      '&[aria-sort="ascending"] .scale-sort-indicator polygon': {
        fill: 'transparent',
      },
      '&:hover[aria-sort="ascending"] .scale-sort-indicator polygon': {
        fill: 'transparent',
      },
      '&:active[aria-sort="ascending"] .scale-sort-indicator polygon': {
        fill: 'transparent',
      },
      '&[aria-sort="descending"] .scale-sort-indicator polygon:first-of-type': {
        fill: 'transparent',
      },
      '&:hover[aria-sort="descending"] .scale-sort-indicator polygon:first-of-type': {
        fill: 'transparent',
      },
      '&:active[aria-sort="descending"] .scale-sort-indicator polygon:first-of-type': {
        fill: 'transparent',
      },
      '&[aria-sort="descending"] .scale-sort-indicator polygon': {
        fill: ({ color }) => color.text,
      },
      '&:hover[aria-sort="descending"] .scale-sort-indicator polygon': {
        fill: ({ color }) => color.primary_hover,
      },
      '&:active[aria-sort="descending"] .scale-sort-indicator polygon': {
        fill: ({ color }) => color.primary_active,
      },
      '& .scale-sort-indicator': {
        position: 'relative',
        display: 'inline-block',
        width: '16px',
        height: '16px',
        margin: '0 0 0 4px',
        '& > svg': {
          position: 'absolute',
          left: 0,
          top: '4px',
          height: '16px',
          width: '16px',
        },
        '& polygon': {
          fill: 'transparent',
        },
      },
    },
  },
};
