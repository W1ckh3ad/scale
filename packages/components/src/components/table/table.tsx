import { Component, Prop, h, Element, Host } from '@stencil/core';
import classNames from 'classnames';

import Base from '../../utils/base-interface';

const name = ' table';
@Component({
  tag: 'scale-table',
  styleUrl: './table.css',
  shadow: false,
})
export class Table implements Base {
  @Element() hostElement: HTMLElement;
  /** (optional) Display sort arrows on/off */
  @Prop() showSort?: boolean = false;
  /** (optional) Visual size */
  @Prop() size?: 'default' | 'small' | string = 'default';
  /** object of the slots in use */
  slots: { header?: Element; table?: Element } = {};

  componentWillLoad() {
    this.hostElement.querySelectorAll('th').forEach(th => {
      th.insertAdjacentHTML(
        'afterbegin',
        `
          <span class="scale-sort-indicator" aria-hidden>
            <svg viewBox="0 0 16 16">
             <polygon transform="translate(8.242641, 10.242641) rotate(45.000000) translate(-8.242641, -10.242641) " points="5.24264069 7.24264069 11.2426407 7.24264069 5.24264069 13.2426407"/></polygon>
             <polygon transform="translate(8.242641, 6.242641) scale(1, -1) rotate(45.000000) translate(-8.242641, -6.242641) " points="5.24264069 3.24264069 11.2426407 3.24264069 5.24264069 9.24264069"/>
            </svg>
          </span>`
      );
    });
  }
  componentWillUpdate() {}
  disconnectedCallback() {}

  render() {
    return (
      <Host class={this.getCssClassMap()}>
        <slot />
      </Host>
    );
  }

  getCssClassMap() {
    return classNames(
      name,
      this.size && `${name}--size${this.size}`,
      this.showSort && `${name}--sortable`
    );
  }
}
