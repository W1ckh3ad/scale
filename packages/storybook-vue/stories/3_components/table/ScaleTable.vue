<template>
  <scale-table :show-sort="showSort" :size="size">
    <slot></slot>
  </scale-table>
</template>

<script>
export default {
  props: {
    showSort: Boolean,
    size: String,
  },
};
</script>
