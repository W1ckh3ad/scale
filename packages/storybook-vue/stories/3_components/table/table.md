# Scale Table

Bacon ipsum dolor amet strip steak andouille short ribs pancetta meatloaf alcatra sausage shankle swine ham hock pork belly corned beef beef cupim sirloin. Tongue sirloin andouille chuck pork biltong bacon ball tip cow cupim.
