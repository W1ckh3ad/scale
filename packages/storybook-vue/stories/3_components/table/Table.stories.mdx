import { Meta, Story, Props, Preview, Description } from '@storybook/addon-docs/blocks';
import ScaleTable from "./ScaleTable.vue";
import description from './table.md'
export const ICON = 'M16.471,8.321 L12.613594,0.502886112 L8.755,8.321 L0.126435258,9.57533801 L6.37,15.66 L4.89610548,24.2548735 L12.613,20.197 L20.3310826,24.2548735 L18.856,15.66 L25.1007528,9.57533801 L16.471,8.321 Z M9.41891527,9.23556376 L12.613,2.761 L15.8082728,9.23556376 L22.951,10.273 L17.7826928,15.3122039 L19.003,22.427 L12.613594,19.067774 L6.223,22.427 L7.44449521,15.3122039 L2.275,10.273 L9.41891527,9.23556376 Z';
import { action } from "@storybook/addon-actions";

<Meta
  title='Components/Table'
  component={ScaleTable}
  argTypes={{
    size: {
      control: { type: 'select', options: ['default', 'small'] }
    },
  }}
/>

export const Template = (args, { argTypes }) => ({
  components: { ScaleTable },
  props: {
    label: String,
    ...ScaleTable.props
  },
  template: `
    <scale-table
      :customClass="customClass"
      :show-sort="showSort"
      :size="size"
      :styles="styles"
    >
      <table id="sortable-table">
        <thead>
          <tr>
            <th width="40%" tabindex="showSort ? 1 : -1" aria-sort="descending" @click="showSort && sortTable('title')" @keyup="showSort && handleKeyUp($event, 'title')" class="title">Title</th>
            <th width="20%" aria-disabled class="tags">Tags</th>
            <th width="20%" tabindex="showSort ? 1 : -1" @click="showSort && sortTable('stats')" @keyup="showSort && handleKeyUp($event, 'stats')" class="stats">Stats</th>
            <th width="10%" tabindex="showSort ? 1 : -1" @click="showSort && sortTable('time')" @keyup="showSort && handleKeyUp($event, 'time')" class="time" style="text-align: right;">Time</th>
            <th width="10%" tabindex="showSort ? 1 : -1" @click="showSort && sortTable('euros')" @keyup="showSort && handleKeyUp($event, 'euros')" class="euros" style="text-align: right;">
              Euros
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Jane</td>
            <td>
              <scale-tag size="small" style="margin-right: 8px;"
                >Other</scale-tag
              >
              <scale-tag size="small" style="margin-right: 8px;">N/A</scale-tag>
              <scale-tag size="small">Demo</scale-tag>
            </td>
            <td>
              <div style="display: inline-flex; align-items: center;">
                <span style="margin-right: 8px;">9.356</span>
                <scale-progress-bar
                  custom-color="#000"
                  variant="info"
                  style="width: 120px;"
                  stroke-width="6"
                  percentage="90"
                />
              </div>
            </td>
            <td style="text-align: right; font-weight: 100;">00:00:20</td>
            <td style="text-align: right; font-weight: 100;">100.245,10</td>
          </tr>
          <tr>
            <td>Jack</td>
            <td>
              <scale-tag size="small" style="margin-right: 8px;"
                >Other</scale-tag
              >
              <scale-tag size="small" style="margin-right: 8px;">N/A</scale-tag>
              <scale-tag size="small">Demo</scale-tag>
            </td>
            <td>
              <div style="display: inline-flex; align-items: center;">
                <span style="margin-right: 8px;">3.356</span>
                <scale-progress-bar
                  custom-color="#000"
                  variant="info"
                  style="width: 120px;"
                  stroke-width="6"
                  percentage="30"
                />
              </div>
            </td>
            <td style="text-align: right; font-weight: 100;">00:00:30</td>
            <td style="text-align: right; font-weight: 100;">100.345,10</td>
          </tr>
          <tr>
            <td>John</td>
            <td>
              <scale-tag size="small" style="margin-right: 8px;"
                >Other</scale-tag
              >
              <scale-tag size="small" style="margin-right: 8px;">N/A</scale-tag>
              <scale-tag size="small">Demo</scale-tag>
            </td>
            <td>
              <div style="display: inline-flex; align-items: center;">
                <span style="margin-right: 8px;">6.356</span>
                <scale-progress-bar
                  custom-color="#000"
                  variant="info"
                  style="width: 120px;"
                  stroke-width="6"
                  percentage="60"
                />
              </div>
            </td>
            <td style="text-align: right; font-weight: 100;">00:00:40</td>
            <td style="text-align: right; font-weight: 100;">100.445,10</td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td>Total</td>
            <td />
            <td />
            <td style="text-align: right;">00:00:20</td>
            <td style="text-align: right;">100.245,10</td>
          </tr>
        </tfoot>
      </table>
    </scale-table>
  `,
  methods: {
    handleKeyUp: function (e, th) {
      if (['Space', 'Enter'].includes(e.code)) {
        this.sortTable(th);
      }
    },
    sortTable: function(className) {
      const getNextSort = sort => {
        if (!sort || ['none', 'other'].includes(sort)) {
          return 'descending';
        }
        if (sort === 'descending') {
          return 'ascending';
        }
        return 'none';
      }
      const ths = document.getElementsByClassName(className)
      const th = ths[ths.length - 1]
      const currentSort = th.getAttribute('aria-sort')
      const nextSort = getNextSort(currentSort)
      const tableHeaders = Array.from(document.getElementsByTagName('TH'))
      const columnIndex = tableHeaders.findIndex(x => x === th)
      tableHeaders.forEach(tableHeader => {
        tableHeader.setAttribute('aria-sort', 'none');
      })
      th.setAttribute('aria-sort', nextSort)
      var table, rows, switching, i, x, y, shouldSwitch
      table = document.getElementById('sortable-table')
      switching = true
      try {
      while (switching) {
        switching = false;
        rows = table.rows;
          for (i = 1; i < rows.length - 1; i++) {
            shouldSwitch = false;
            x = rows[i].getElementsByTagName('TD')[columnIndex];
            y = rows[i + 1].getElementsByTagName('TD')[columnIndex];
            if (
              !['ascending', 'descending'].includes(nextSort) ||
              x.parentElement.parentElement.tagName === 'TFOOT' ||
              y.parentElement.parentElement.tagName === 'TFOOT'
            ) {
              break;
            }
            if (
              (nextSort === 'descending' ? y : x).innerHTML.toLowerCase() >
              (nextSort === 'descending' ? x : y).innerHTML.toLowerCase()
            ) {
              shouldSwitch = true;
              break;
            }
          }
          if (shouldSwitch) {
            rows[i].parentNode.insertBefore(rows[i + 1], rows[i]);
            switching = true;
          }
        }
      } catch(error){}
    }
  }
})

<Description>{description}</Description>

<Preview withSource="none">
  <Story
    name='Standard'
    args={{}}
  >
    {Template.bind({})}
  </Story>
</Preview>

<Props story='Standard' />


## Small

<Preview withSource="none">
  <Story
    name='Small'
    args={{
      size: 'small',
    }}
  >
    {Template.bind({})}
  </Story>
</Preview>

```html
<scale-table size="small">
  <table>
    <h5>Table title</h5>
    <thead>
      <tr>
        <th>Title</th>
        <th>Tags</th>
        <th>Stats</th>
        <th>Time</th>
        <th>Euros</th>
      </tr>
    </thead>
    <tbody>
      ...
    </tbody>
    <tfoot>
      <tr>
        <td>Total</td>
        <td></td>
        <td></td>
        <td>00:00:20</td>
        <td>100.245,10</td>
      </tr>
    </tfoot>
  </table>
</scale-table>
```


## With Sorting Icons

Please switch to the Canvas view for a working example

<Preview withSource="none">
  <Story
    name='With Sorting Icons'
    args={{
      showSort: true,
    }}
  >
    {Template.bind({})}
  </Story>
</Preview>


```html
<scale-table show-sort>
  <table>
    <h5>Table title</h5>
    <thead>
      <tr>
        <!-- aria-sort can be descending or ascending -->
        <th tabindex="0" aria-sort="descending" @keyup="handleKeyUp($event, 'title')" @click="sortTable('title')" id="title">Title</th>
        <!-- aria-disabled disables sorting -->
        <th aria-disabled>Tags</th>
        <th tabindex="0" @keyup="handleKeyUp($event, 'stats')" @click="sortTable('stats')" id="stats">Stats</th>
        <th tabindex="0" @keyup="handleKeyUp($event, 'time')" @click="sortTable('time')" id="time">Time</th>
        <th tabindex="0" @keyup="handleKeyUp($event, 'euros')" @click="sortTable('euros')" id="euros">
          Euros
        </th>
      </tr>
    </thead>
    <tbody>
      ...
    </tbody>
    <tfoot>
      <tr>
        <td>Total</td>
        <td></td>
        <td></td>
        <td>00:00:20</td>
        <td>100.245,10</td>
      </tr>
    </tfoot>
  </table>
</scale-table>
```
