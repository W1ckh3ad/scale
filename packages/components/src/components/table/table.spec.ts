import { newSpecPage } from '@stencil/core/testing';
import { Table } from './table';
describe('Table', () => {
  let element;

  beforeEach(async () => {
    element = new Table();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Table],
      html: `
      <scale-table>
        <div>Table header</div>
        <table>
          <thead>
            <tr>
              <th>Title</th>
              <th>Time</th>
              <th>Euros</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>University of Plymouth</td>
              <td>00:00:20</td>
              <td>100.245,10</td>
            </tr>
          </tbody>
          <tfoot>
            <tr>
              <td>Total</td>
              <td />
              <td />
              <td>00:00:20</td>
              <td>100.245,10</td>
            </tr>
          </tfoot>
        </table>
      </scale-table>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.size = 'big';
    expect(element.getCssClassMap()).toContain('table--size-big');
  });
});
