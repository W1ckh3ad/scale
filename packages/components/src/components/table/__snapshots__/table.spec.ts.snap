// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should match snapshot 1`] = `
<scale-table class="table--size-default-0-0-5 table-0-0-4">
  <!---->
  <div>
    Table header
  </div>
  <table>
    <thead>
      <tr>
        <th>
          <span class="scale-sort-indicator">
            <svg height="24px" viewBox="0 0 24 24" width="24px">
              <polygon points="11.8284271 16.6568542 14.6568542 13.8284271 9 13.8284271"></polygon>
              <polygon points="11.8284271 8 14.6568542 10.8284271 9 10.8284271"></polygon>
            </svg>
          </span>
          Title
        </th>
        <th>
          <span class="scale-sort-indicator">
            <svg height="24px" viewBox="0 0 24 24" width="24px">
              <polygon points="11.8284271 16.6568542 14.6568542 13.8284271 9 13.8284271"></polygon>
              <polygon points="11.8284271 8 14.6568542 10.8284271 9 10.8284271"></polygon>
            </svg>
          </span>
          Time
        </th>
        <th>
          <span class="scale-sort-indicator">
            <svg height="24px" viewBox="0 0 24 24" width="24px">
              <polygon points="11.8284271 16.6568542 14.6568542 13.8284271 9 13.8284271"></polygon>
              <polygon points="11.8284271 8 14.6568542 10.8284271 9 10.8284271"></polygon>
            </svg>
          </span>
          Euros
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          University of Plymouth
        </td>
        <td>
          00:00:20
        </td>
        <td>
          100.245,10
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr>
        <td>
          Total
        </td>
        <td></td>
        <td></td>
        <td>
          00:00:20
        </td>
        <td>
          100.245,10
        </td>
      </tr>
    </tfoot>
  </table>
</scale-table>
`;
